src/components/SurveyCourseList.tsx

```diff
@@ -0,0 +1,173 @@
+import { gql, useQuery } from "@apollo/client";
+import {
+  Divider,
+  Flex,
+  FormLabel,
+  Grid,
+  GridItem,
+  Heading,
+  Radio,
+  RadioGroup,
+  Stack,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { CourseInterface } from "../pages/Survey";
+
+const COURSES = gql`
+  query GetCourses {
+    survey {
+      courses {
+        subject
+        code
+        term
+      }
+    }
+  }
+`;
+
+interface PreferenceInterface {
+  subject: string;
+  code: string;
+  term: string;
+  able: string;
+  willing: string;
+}
+
+interface PreferenceListInterface {
+  [key: string]: PreferenceInterface;
+}
+
+interface ChildProps {
+  handleCourseChange(course: CourseInterface, value: number): void;
+}
+
+export const SurveyCourseList: React.FC<ChildProps> = (props) => {
+  const { loading, error, data } = useQuery(COURSES, {});
+  const [preferences, setPreferences] = useState<PreferenceListInterface>({});
+
+  useEffect(() => {
+    if (typeof error != "undefined") {
+      console.log(error);
+    }
+  }, [error]);
```

Comment:
Should probably remove debugging stuff before merging

Reply:
Got rid of this